deploy/cloudfront.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 The Hugo Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deploy
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudfront"
+)
+
+// InvalidateCloudFront invalidates the CloudFront cache for distributionID.
+// It uses the default AWS credentials from the environment.
+func InvalidateCloudFront(ctx context.Context, distributionID string) error {
+	// SharedConfigEnable enables loading "shared config (~/.aws/config) and
+	// shared credentials (~/.aws/credentials) files".
+	// See https://docs.aws.amazon.com/sdk-for-go/api/aws/session/ for more
+	// details.
+	sess, err := session.NewSessionWithOptions(session.Options{SharedConfigState: session.SharedConfigEnable})
```

Comment:
Is this the same authentication I would get for the S3 upload step? I say this because I find it a little surprising that the Google Cloud library doesn't have CDN as a concept. It would be ideal if we could share session/at least use the same code path. But we should at least make sure that if 1) S3 upload is OK 2) CDN invalidate should not fail because of an authentication failure.

Reply:
> Is this the same authentication I would get for the S3 upload step?

Yes: https://github.com/google/go-cloud/blob/master/blob/s3blob/s3blob.go#L106-L110.

> It would be ideal if we could share session/at least use the same code path.

I filed an issue for Go CDK to add a helper for this so we can share the code:
https://github.com/google/go-cloud/issues/2003

> But we should at least make sure that if 1) S3 upload is OK 2)
> CDN invalidate should not fail because of an authentication failure.

Given that the code path is the same, this should be true.

Reply:
OK.